src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -137,73 +114,83 @@ export const SEQUENTIAL_COLOR_SCHEMES = ['blues', 'greens', 'greys', 'purples',
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'range' || nestedProp === 'domain') {
-      return generateSchema(prop, nestedProp, isDiscrete(fieldDef) ?
-        DEFAULT_DISCRETE_SCALE_ARRAY_UISCHEMA : DEFAULT_CONTINUOUS_SCALE_ARRAY_UISCHEMA, propTab, baseUISchema);
-    }
-  } else if (prop === 'axis') {
-    if (nestedProp === 'orient') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: channel === 'y' ? ['left', 'right'] : ['top', 'bottom']
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'title') {
-      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-    }
-  } else if (prop === 'stack') {
-    const propertySchema = {
-      ...baseUISchema,
-      enum: (vlSchema as any).definitions.StackOffset.enum,
-    };
-    return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-  } else if (prop === 'size') {
-    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-  } else if (prop === 'legend') {
-    if (nestedProp === 'orient') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: (vlSchema as any).definitions.LegendOrient.enum
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    } else if (nestedProp === 'title') {
-      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-    } else if (nestedProp === 'type') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: (vlSchema as any).definitions.Legend.properties.type.enum,
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    }
-  } else if (prop === 'format') {
-    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-  } else {
-    throw new Error('Property combination not recognized');
+      return generateScaleEditorSchema(nestedProp, scaleTypes, fieldDef, baseUISchema);
+    case 'axis':
+      return generateAxisEditorSchema(nestedProp, channel as Channel, baseUISchema, propertyKey);
+    case 'stack':
+      return generateStackEditorSchema(baseUISchema);
+    case 'legend':
+      return generateLegendEditorSchema(nestedProp, baseUISchema, propertyKey);
+    case 'format':
+      return generateFormatEditorSchema(baseUISchema);
+    default:
+      throw new Error('Property combination not recognized');
+  }
+}
+
+function generateFormatEditorSchema(baseUISchema: SchemaProperty) {
+  return generateTextBoxSchema('format', '', baseUISchema, 'string');
+}
+
+function generateLegendEditorSchema(nestedProp: string, baseUISchema: SchemaProperty, propertyKey: string) {
```

Comment:
nestedProp => `legendProp: keyof Legend` - -same for scale/axis methods.

Reply:
This makes sense. When I call this method (and the other subsequent methods), how do I make it so that `nestedProp`, which is a string, is a `keyof Legend`, for instance. I had to cast, but I feel like that is not the right way to do it.